### PR TITLE
POST | /locations

### PR DESCRIPTION
**Pull request by Alaalser**
Relates #79
### ِAdd new location
- Doing the queries 
- Creating the function 
- Handling errors
- Creating the tests

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -243,14 +243,38 @@ describe('Get all kindergartens', () => {
 });
 
 // test the route /locations
-describe('Get locations', () => {
-  test('Route /users status 200, json header', async () => {
+describe('Locations', () => {
```
Comment (Mu7ammadAbed) [inline test/index.test.js:246]: split the tests for different routes into different describes

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/controllers/locations/postLocations/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const { postLocationsQuery } = require('../../../database/queries');
+const { locationSchema } = require('../../../utils/validation');
+const { boomify } = require('../../../utils');
+
+const postLocations = async (req, res, next) => {
+  try {
+    const { subLocation, mainLocation } = req.body;
+    await locationSchema.validate(
+      { subLocation, mainLocation },
```
Comment (Mu7ammadAbed) [inline server/controllers/locations/postLocations/index.js:9]:
```suggestion
    const { subLocation, mainLocation } = await locationSchema.validate(
      req.body,
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: test/index.test.js [hunk 3]
```diff
@@ -243,14 +243,38 @@ describe('Get all kindergartens', () => {
...
+  });
+
+  test('Route get /locations status 200, json header', async () => {
     expect.assertions(1);
     const res = await request(app)
       .get('/api/v1/locations')
       .expect(200)
       .expect('Content-Type', /json/);
-    expect(res.body.data).toHaveLength(23);
+    expect(res.body.data).toHaveLength(24);
```
Comment (Mu7ammadAbed) [inline test/index.test.js:277]: Return this to `23` again and edit `beforeAll` to `beforeEach` and fix the other test which will fail
```suggestion
    expect(res.body.data).toHaveLength(23);
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]
You still didn't edit the `beforeEach` note and fix the tests

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Osama-you — APPROVED** [review]

**Review by alaa-sahloub — APPROVED** [review]

**Review by Israa91 — APPROVED** [review]

**Pull request merged**